Replacement PR: Specify max tip on each tx

cliffhall opened the pull request:
## Replacement PR. Supersedes #256  

- In .env.example
  - Changed comment identifiers to # from // (this parsed by shell)
  - added `MAX_TIP`

- In deploy.ts,
  - in DeploymentExecutor
    - added `maxTip` and `txOptions`
    - in constructor
      - set maxTip to process.env.MAX_TIP
      - set txOptions to an object mapping `maxPriorityFeePerGas `to `this.maxTip`
    - in all deploy and method calls, pass `this.txOptions` as last or sole argument
        
- This fixes #255

cliffhall committed:
Specify max tip on each tx

- In .env.example
  - Changed comment identifiers to # from // (this parsed by shell)
  - added MAX_TIP

- In deploy.ts,
  - in DeploymentExecutor
    - added maxTip and txOptions
    - in constructor
      - set maxTip to process.env.MAX_TIP
      - set txOptions to an object mapping maxPriorityFeePerGas to this.maxTip
      - in all deploy and method calls, pass this.txOptions as last or sole argument
set maxTip to process.env.MAX_TIP
set txOptions to an object mapping maxPriorityFeePerGas to this.maxTip
in all deploy and method calls, pass this.txOptions as last or sole argument

cliffhall commented:
This is a replacement for PR #256. I bungled the branch merging from main (there were conflicts so I tried to revert the merge and defer reconciliation but it didn't go so well) and and since there were only two files, I just made a new branch and replacement PR.

hswopeams commented on .env.example:
```diff
@@ -1,16 +1,23 @@
 CONDITIONAL_COMMIT_TOKEN_METADATA_URI=changeme
-VOUCHERSETS_METADATA_URI=changeme
-VOUCHERS_METADATA_URI=changeme
+METADATA_BASE=changeme
```
These still aren't correct. It should be 
OUCHERSETS_METADATA_URI=changeme
VOUCHERS_METADATA_URI=changme

METADATA_BASE should be removed

cliffhall replied:
Done.

hswopeams reviewed (changes requested):
See comment about .env.example

cliffhall committed:
Environment vars

- Responding to Heather's change requests on the PR

* In env.example,
  - add VOUCHERSETS_METADATA_URI
  - add VOUCHERS_METADATA_URI
  - remove METADATA_BASE
  - add comment grouping routes

cliffhall committed:
Environment vars

- Responding to Heather's change requests on the PR

* In env.example,
  - remove ERC1155_ROUTE
  - remove ERC721_ROUTE

cliffhall committed:
Merge branch 'main' into deployment-gas-redux

hswopeams commented:
Approved, assuming tests will pass.

mischat reviewed (approved):
Let's do this.
